### DISCUSSION TITLE
How can a user protect himself from potential malicious software in homebrew?

### DISCUSSION THREAD

**Discussion by jchkn** [Homebrew/discussions#2977] [Everyday usage]
From what I've seen, it seems the maintainers delegate security considerations mostly to the enduser, or is this a wrong impression? This is also okay, just wonder what a user can do, besides trust?

**Comment by neutric** [comment]
> From what I've seen, it seems the maintainers delegate security considerations mostly to the enduser, or is this a wrong impression?

Can you elaborate?

> This is also okay, just wonder what a user can do, besides trust?

Audit before upgrading a formula/cask. Sharing the fruits of your labor should you unearth something concerning. Also, contribute by keeping formulae/casks current and help enabling best practices such as checksumming (by changing upstream developers stance on issues such as retagging releases or offering versioned software downloads instead of just `latest`).

**Comment by jchkn** [comment]
> Can you elaborate?

For example I read in "Acceptable Casks":

**Homebrew Cask has long decided it will not be an active gatekeeper ([macOS already has one](https://support.apple.com/en-us/HT202491)) and [users are expected to know about the software they are installing](https://docs.brew.sh/Acceptable-Casks#homebrew-cask-is-not-a-discoverability-servic_e).**

This is about Homebrew Cask, in "Acceptable Formulae" "security vulnerabilities" are only mentioned in a chapter about niche software. There are a few other places where I have gained the impression that security is being handled reactively, but not proactively.

I can understand why that is, I'm just wondering what options we have to improve security, not as a contributor but as a user. For example, can all Formulas (bottles)  consistently be verified using reproducible-builds? Is there any way to restrict the read/write permissions from the commandos to specific directories (besides typical unix-user rights management).  
That is what my question is about, practical tips to increase the security.

**Reply by gromgit** [reply]
As a Homebrew _user_, there's really only two things you _can_ do:
1. Don't install what you don't need.
2. Keep what you installed up-to-date (`brew update`/`brew upgrade` regularly).
